components/activity_and_insights/insights/top_threads/top_threads.tsx

```diff
@@ -0,0 +1,169 @@
+    }, [getTopTeamThreads]);
+
+    const getMyTeamThreads = useCallback(async () => {
+        if (props.filterType === InsightsScopes.MY) {
+            setLoading(true);
+            const data: any = await dispatch(getMyTopThreads(currentTeamId, 0, 3, props.timeFrame));
+            if (data.data?.items) {
+                setTopThreads(data.data.items);
+            }
+            setLoading(false);
+        }
+    }, [props.timeFrame, props.filterType]);
+
+    useEffect(() => {
+        getMyTeamThreads();
+    }, [getMyTeamThreads]);
+
+    const skeletonLoader = useMemo(() => {
+        const entries = [];
+        for (let i = 0; i < 3; i++) {
+            entries.push(
+                <div
+                    className='top-thread-loading-container'
+                    key={i}
+                >
+                    <div className='top-thread-loading-row'>
+                        <CircleLoader
+                            size={20}
+                        />
+                        <TitleLoader/>
+                    </div>
+                    <div className='loading-lines'>
+                        <TitleLoader/>
+                        <TitleLoader/>
+                    </div>
+                </div>,
+            );
+        }
+        return entries;
+    }, []);
+
+    const openRHS = async (post: Post) => {
+        dispatch(selectPost(post));
+    };
+
+    return (
+        <div className='top-thread-container'>
+            {
+                loading &&
+                skeletonLoader
+            }
+            {
+                (topThreads && !loading) &&
+                <div className='thread-list'>
+                    {
+                        topThreads.map((thread, key) => {
+                            return (
+                                <div
+                                    className='thread-item'
+                                    onClick={() => {
+                                        openRHS(thread.post);
+                                    }}
+                                    key={key}
+                                >
+                                    <div className='thread-details'>
+                                        <Avatar
+                                            url={imageURLForUser(thread.user_id)}
+                                            size={'xs'}
+                                        />
+                                        <span className='display-name'>{displayUsername(thread.user_information as UserProfile, teammateNameDisplaySetting)}</span>
+                                        <Badge>
+                                            {thread.channel_display_name}
+                                        </Badge>
+                                        <div className='reply-count'>
+                                            <i className='icon icon-reply-outline'/>
+                                            <span>{thread.reply_count}</span>
```

Comment:
https://github.com/mattermost/mattermost-server/pull/20195#issuecomment-1144737818 Other things LGTM, only fixes related to the removal of  these fields remain.